ceno_emul/src/platform.rs

```diff
@@ -9,33 +9,42 @@ use crate::addr::{Addr, RegIdx};
 /// - codes of environment calls.
 #[derive(Clone, Debug)]
 pub struct Platform {
-    pub rom: Range<Addr>,
-    pub ram: Range<Addr>,
+    pub prog_code: Range<Addr>,
+    pub prog_data: Option<HashSet<Addr>>,
```

Comment:
Consider changing to `BTreeSet` which has `pub const fn new() -> BTreeSet<T>` to make `const` objects.

Reply:
Done.

Reply:
Interesting.  Thanks!

Then a `HashSet` is as good as the `BTreeSet`, but please use a doc comment for the explanation, so it shows up in our tooling.  Thanks!